src/agent/coverage/src/pe.rs

```diff
@@ -289,6 +280,78 @@ pub fn process_module(
     anyhow::bail!("PE missing codeview pdb debug info")
 }
 
+fn find_pdb_path(pe_path: &Path, cv: &CodeviewPDB70DebugInfo) -> Result<PathBuf> {
+    let cv_filename = CStr::from_bytes_with_nul(cv.filename)?.to_str()?;
+
+    // This field is named `filename`, but it may be an absolute path.
+    // The callee `sym_find_pdb_file_in_path()` handles either.
+    let cv_filename = Path::new(cv_filename);
+
+    // Any nonzero value (cast to a `HANDLE`) should work here.
+    //
+    // The docs warn that we must not intialize the dbghelp symbol handler
+    // with the current process handle when debugging another process. When
+    // we invoke this function, we will often be doing exactly this.
+    //
+    // Assume this is only relevant when making queries about some debuggee's
+    // address space, which we will not do here.
+    let handle = win_util::process::current_process_handle();
+
+    let dbghelp = debugger::dbghelp::lock()?;
+    dbghelp.sym_initialize(handle)?;
+    let _cleanup = DbgHelpCleanupGuard::new(&dbghelp, handle);
+
+    // Enable signature and age checking.
+    let options = dbghelp.sym_get_options();
+    dbghelp.sym_set_options(options | SYMOPT_EXACT_SYMBOLS);
+
+    let mut search_path = dbghelp.sym_get_search_path(handle)?;
+
+    log::debug!("initial search path = {:?}", search_path);
+
+    // Try to add the directory of the PE to the PDB search path.
```

Comment:
Is this really necessary? I'm thinking onefuzz should be setting the symbol path for other reasons, e.g. the input tester needs the correct symbol path to produce good call stacks, and also if the user remotes in to debug a crash, the symbol path should be set correctly for them already.

For the "local" scenario where we run coverage on the same machine the target is built on, the full path in the PE should be all we need.

Reply:
>  I'm thinking onefuzz should be setting the symbol path for other reasons

Agreed. The goal here was to make this code path more robust in various contexts. This is inessential and we could remove it.

> [...] the full path in the PE should be all we need.

Maybe. I found that this was painfully brittle in practice. In local fuzzing, I would actually expect that the user typically would NOT build in the same directory the target would be invoked from. But again, OneFuzz should really be setting `_NT_SYMBOL_PATH` to include the setup dir, which will handle almost all cases. However, the instant a user nests a target module in the setup dir, it will then break coverage.

The original inspiration was actually item (3) [here](https://docs.microsoft.com/en-us/windows/win32/debug/symbol-paths#specifying-search-paths), which appears to actually be _false_, and in need of correction. The actual behavior is described by the docs for the [`UserSearchPath` parameter](https://docs.microsoft.com/en-us/windows/win32/api/dbghelp/nf-dbghelp-syminitializew#parameters) of `SymInitializeW`. That said, IMO, the incorrectly-described behavior is more intuitive.